### PR TITLE
Update MSRV to 1.66.1

### PR DESCRIPTION
**Pull request by Velfi**
## Motivation and Context

#2467 

## Description

Updates the MSRV to 1.66.1

## Testing

ran existing tests

## Checklist

- [x] I have updated `CHANGELOG.next.toml` if I made changes to the smithy-rs codegen or runtime crates
- [x] I have updated `CHANGELOG.next.toml` if I made changes to the AWS SDK, generated SDK code, or SDK runtime crates

----

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

File: rust-runtime/aws-smithy-checksums/src/lib.rs [hunk 1]
```diff
@@ -64,11 +64,11 @@ impl ChecksumAlgorithm {
     /// Return the `HttpChecksum` implementor for this algorithm
     pub fn into_impl(self) -> Box<dyn http::HttpChecksum> {
         match self {
-            Self::Crc32 => Box::new(Crc32::default()),
-            Self::Crc32c => Box::new(Crc32c::default()),
-            Self::Md5 => Box::new(Md5::default()),
-            Self::Sha1 => Box::new(Sha1::default()),
-            Self::Sha256 => Box::new(Sha256::default()),
+            Self::Crc32 => Box::<Crc32>::default(),
```
Comment (rcoh) [inline rust-runtime/aws-smithy-checksums/src/lib.rs:67]: lol thanks clippy but that is definitely not better

File: rust-runtime/aws-smithy-http-server/src/protocols.rs [hunk 2]
```diff
@@ -21,6 +22,7 @@ pub fn content_type_header_empty_body_no_modeled_input(headers: &HeaderMap) -> R
     }
 }
 
+#[allow(clippy::result_large_err)]
```
Comment (rcoh) [inline rust-runtime/aws-smithy-http-server/src/protocols.rs:25]: since this is hand written, cc @awslabs/smithy-rs-server we might want to look into that

**Review by rcoh — APPROVED** [review]
I think we also need to change the default list of clippy lints in kotlin

File: rust-runtime/aws-smithy-checksums/src/lib.rs (hunk 1 above)
Comment (Velfi) [reply rust-runtime/aws-smithy-checksums/src/lib.rs:67]: Yeah, I'm getting a bit annoyed at Clippy. The last few MSRV updates have introduced some really pedantic lints.

**Review by hlbarber — APPROVED** [review]

File: rust-runtime/aws-smithy-http-server/src/protocols.rs (hunk 2 above)
Comment (hlbarber) [reply rust-runtime/aws-smithy-http-server/src/protocols.rs:25]: Ticketed this up https://github.com/awslabs/smithy-rs/issues/2470

**Comment by Velfi** [comment]
I had to fix several tests that were failing due to creating a module named "lib". Those changes need the most review. Affected modules are the structure and enum generator tests and a recursive struct test.

**Comment by Velfi** [comment]
If you approve this PR, then please merge it

File: rust-runtime/aws-smithy-http/src/body.rs [hunk 3]
```diff
@@ -109,13 +113,16 @@ impl SdkBody {
         }
     }
 
+    /// When an SdkBody is read, the inner data must be consumed. In order to do this, the SdkBody
```
Comment (jdisanti) [inline rust-runtime/aws-smithy-http/src/body.rs:116]: Thank you for adding docs!

**Review by jdisanti — APPROVED** [review]

**Pull request merged**